src/Google/Client.php

```diff
@@ -795,7 +795,7 @@ public function execute(RequestInterface $request, $expectedClass = null)
     // this is where most of the grunt work is done
     $http = $this->authorize();
 
-    return Google_Http_REST::execute($http, $request, $expectedClass, $this->config['retry']);
+    return Google_Http_REST::execute($http, $request, $expectedClass, null, $this->config['retry']);
```

Comment:
Thanks for the PR @Yahav :)

From what I can tell, it looks like the `retry` key of the client configuration should map [here](https://github.com/google/google-api-php-client/blob/master/src/Google/Http/REST.php#L44), so the original code looks correct. In order to provide custom retry maps, I believe what we need to do is expose a `retryMap` setting in the `Google_Client` constructor which we can pass down into `Google_Http_REST` like so:

```php
return Google_Http_REST::execute($http, $request, $expectedClass, $this->config['retry'], $this->config['retryMap']);
```

WDYT?

Reply:
The original code isn't working. please read the following carefully:
https://github.com/google/google-api-php-client/issues/1487
the client config 'retry' is documented as:

```
 // Task Runner retry configuration
          // @see Google_Task_Runner
          'retry' => array(),
```

however, no retry config is used at the Google_Task_Runner, there is a 'retries' key but not retry.
this:

```
return Google_Http_REST::execute($http, $request, $expectedClass, $this->config['retry'], $this->config['retryMap']);
```

won't work as an array is expected..
It really seem like the whole task runner wasn't implemented correctly since basic functionality is missing, for instance, you can't define custom retryMap nor can you set the $maxAttempts (which defaults to 1 so basically even the default retryMap won't function correctly since 

```
 /**
   * Checks if a retry can be attempted.
   *
   * @return boolean
   */
  public function canAttempt()
  {
    return $this->attempts < $this->maxAttempts;
  }
```

it will always return true when the maxAttempts is 1, and lacking a way to set it..makes it useless.
the following was written in order to set maxAttempts using a config var but it was never passed.

```
if (isset($config['retries'])) {
      if ($config['retries'] < 0) {
        throw new Google_Task_Exception(
            'Task configuration `retries` must not be negative.'
        );
      }
      $this->maxAttempts += $config['retries'];
    }
```

Anyway, its quite a mess, i'm gonna give it a try and adjust whatever i can to make it functional, i'm also testing this in realtime..

Reply:
Done, should be alright now.  can you confirm?